R/validityChecks.R

```diff
@@ -999,3 +999,165 @@
         }
     }
 }
+
+.valid.detectSpatialContext.input <- function(object,
+                                              entry,
+                                              threshold,
+                                              name){
+  if (!is(object, "SingleCellExperiment")) {
+    stop("'object' needs to be a SingleCellExperiment object.")
+  }
+  if (!entry %in% names(colData(object))) {
+    stop("'entry' not in 'colData(object)'.")
+  }
+  if (!(is.numeric(threshold) & (0 <= threshold && threshold <= 1)) 
+  ) {
+    stop("'threshold' needs to be a single numeric between 0-1.")
+  }
+  if (length(name) != 1 | !is.character(name)) {
+    stop("'name' has to be a single character'.")
+  }
+}
+
+.valid.filterSpatialContext.input <- function(object, 
+                                              entry, 
+                                              group_by,
+                                              group_threshold,
+                                              cells_threshold, 
+                                              name){
+  
+  if (!is(object, "SingleCellExperiment")) {
+    stop("'object' needs to be a SingleCellExperiment object.")
+  }
+  
+  if (!entry %in% names(colData(object))) {
+    stop("'entry' not in 'colData(object)'.")
+  }
+  
+  if (!group_by %in% names(colData(object))) { 
+    stop("'group_by' not in 'colData(object)'.")
+  }
+  
+  if (!is.null(group_threshold) &&
```

Comment:
In theory you would also need to test if the entry is a single entry. But I think it's fine.